[2024-06-07 16:06:18,170][__main__][INFO] - data loaded
[2024-06-07 16:06:18,189][__main__][INFO] - Latest Local Model download
[2024-06-07 16:06:18,189][root][INFO] - downloaded local_model_name: SMILESModel_local_model_V1.pth
[2024-06-07 16:06:18,191][root][INFO] - init model_type: Pytorch
[2024-06-07 16:07:40,976][root][INFO] - bulid model
[2024-06-07 16:07:40,976][root][INFO] - FL start
[2024-06-07 16:07:40,986][fedops.client.client_api][INFO] - FL_server_IP:port - ccl.gachon.ac.kr:40021
[2024-06-07 16:07:40,987][root][INFO] - FL learning ready
[2024-06-07 16:07:40,987][root][INFO] - fl_task_id: admet2
[2024-06-07 16:07:40,987][root][INFO] - dataset: carboxylic
[2024-06-07 16:07:40,987][root][INFO] - output_size: 1
[2024-06-07 16:07:40,987][root][INFO] - validation_split: 0.2
[2024-06-07 16:07:40,988][root][INFO] - model_type: Pytorch
[2024-06-07 16:07:40,988][root][INFO] - wandb_key: ce53e10675102394f9c8ac9f89c76bcbf14ea805
[2024-06-07 16:07:40,988][root][INFO] - wandb_account: z8086486
[2024-06-07 16:07:54,129][root][INFO] - label_values: [[0, 0]]
[2024-06-07 16:07:54,129][flwr][WARNING] - DEPRECATED FEATURE: flwr.client.start_numpy_client() is deprecated. 
	Instead, use `flwr.client.start_client()` by ensuring you first call the `.to_client()` method as shown below: 
	flwr.client.start_client(
		server_address='<IP>:<PORT>',
		client=FlowerClient().to_client(), # <-- where FlowerClient is of type flwr.client.NumPyClient object
	)
	Using `start_numpy_client()` is deprecated.

            This is a deprecated feature. It will be removed
            entirely in future versions of Flower.
        
[2024-06-07 16:07:54,142][flwr][DEBUG] - Opened insecure gRPC connection (no certificates were passed)
[2024-06-07 16:07:54,143][flwr][DEBUG] - ChannelConnectivity.IDLE
[2024-06-07 16:07:54,145][flwr][DEBUG] - ChannelConnectivity.CONNECTING
[2024-06-07 16:07:54,149][flwr][DEBUG] - ChannelConnectivity.READY
[2024-06-07 16:07:54,217][flwr][INFO] - 
[2024-06-07 16:07:54,218][flwr][INFO] - [RUN 0, ROUND ]
[2024-06-07 16:07:54,218][flwr][INFO] - Received: train message 729652e1-fbcf-4c7f-944f-42b133ac4aee
[2024-06-07 16:07:54,319][flwr][ERROR] - Client raised an exception.
Traceback (most recent call last):
  File "/home/ccl/anaconda3/envs/isenv/lib/python3.11/site-packages/flwr/client/app.py", line 496, in _start_client_internal
    reply_message = client_app(message=message, context=context)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ccl/anaconda3/envs/isenv/lib/python3.11/site-packages/flwr/client/client_app.py", line 98, in __call__
    return self._call(message, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ccl/anaconda3/envs/isenv/lib/python3.11/site-packages/flwr/client/client_app.py", line 81, in ffn
    out_message = handle_legacy_message_from_msgtype(
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ccl/anaconda3/envs/isenv/lib/python3.11/site-packages/flwr/client/message_handler/message_handler.py", line 130, in handle_legacy_message_from_msgtype
    fit_res = maybe_call_fit(
              ^^^^^^^^^^^^^^^
  File "/home/ccl/anaconda3/envs/isenv/lib/python3.11/site-packages/flwr/client/client.py", line 234, in maybe_call_fit
    return client.fit(fit_ins)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/ccl/anaconda3/envs/isenv/lib/python3.11/site-packages/flwr/client/numpy_client.py", line 238, in _fit
    results = self.numpy_client.fit(parameters, ins.config)  # type: ignore
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ccl/anaconda3/envs/isenv/lib/python3.11/site-packages/fedops/client/client_fl.py", line 135, in fit
    trained_model = self.train_torch(self.model, self.train_loader, self.val_loader, epochs)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ccl/Desktop/isfolder/issrc/fedops/FedOps-ADMET2/models.py", line 56, in custom_train_torch
    if val_loss < best_val_loss:
                  ^^^^^^^^^^^^^
UnboundLocalError: cannot access local variable 'best_val_loss' where it is not associated with a value
[2024-06-07 16:07:54,352][flwr][DEBUG] - gRPC channel closed
[2024-06-07 16:07:54,370][root][INFO] - notify_fail start
[2024-06-07 16:07:54,391][root][ERROR] - trainFin
[2024-06-07 16:10:10,484][root][INFO] - ccl-y-server;f0:2f:74:96:d6:64-client close
